### PR TITLE
reset incentives.PendingMultiRewards and rewards.SharesAndWithdrawnRewards for Pool aUSD-IBTC

### PR DESCRIPTION
**Pull request by wangjj9219**

### CODE REVIEW

**Review by xlc — DISMISSED** [review]

**Comment by wangjj9219** [comment]
I think this migration should reset rewards.PoolInfos for pool aUSD-IBTC, I will update code

**Review by zjb0807 — DISMISSED** [review]

File: modules/incentives/src/migration.rs [hunk 1]
```diff
@@ -19,29 +19,91 @@
...
+
+		let mut total_share: Balance = Default::default();
+
+		// reset SharesAndWithdrawnRewards
+		for (who, (share, _)) in orml_rewards::SharesAndWithdrawnRewards::<T>::iter_prefix(&pool_id) {
+			orml_rewards::SharesAndWithdrawnRewards::<T>::mutate(&pool_id, &who, |(_, withdrawn_rewards)| {
+				*withdrawn_rewards = WithdrawnRewards::new();
+			});
+
+			total_share = total_share.saturating_add(share);
```
Comment (zjb0807) [inline modules/incentives/src/migration.rs:78]: The WithdrawnRewards is BTreeMap, e.g.
```
{
        {token:ACA}: 2,047,606,129,239,216
        {token:AUSD}: 62,077,309,016,549,947,911
        {foreignAsset:4}: 63,305,740,673,477
}
```

But the total_share is Balance, I'm not sure the result is correct.

Comment (zjb0807) [reply modules/incentives/src/migration.rs:78]: https://github.com/open-web3-stack/open-runtime-module-library/blob/master/rewards/src/lib.rs#L80

Comment (wangjj9219) [reply modules/incentives/src/migration.rs:78]: here is vaule definition for `SharesAndWithdrawnRewards` : 
https://github.com/open-web3-stack/open-runtime-module-library/blob/44fda4432b71f49ee59a650cf4775db895ab97af/rewards/src/lib.rs#L110

The previous one of the tuple is `Balance` type to record the staked LP token amount,  `WithdrawnRewards` records the withdrawn rewards for calculation unsettled rewards. 

And `rewards.PoolInfos` for aUSD-IBTC has been killed in before referendum. Therefore, in addition to clearing the unsettled rewards(`WithdrawnRewards` in `SharesAndWithdrawnRewards` ) and the rewards to be claimed(incentives.PendingMultiRewards), the total amount of staked LP token should be counted and reset in `rewards.PoolInfos`.

**Review by zjb0807 — DISMISSED** [review]

File: modules/incentives/src/migration.rs [hunk 2]
```diff
@@ -19,29 +19,91 @@
...
+	fn post_upgrade() -> Result<(), &'static str> {
+		let pool_id = GetPoolId::get();
+		let mut total_share = Balance::default();
+
+		for (_, (share, withdrawn_rewards)) in orml_rewards::SharesAndWithdrawnRewards::<T>::iter_prefix(&pool_id) {
+			assert_eq!(withdrawn_rewards, WithdrawnRewards::new());
+			total_share = total_share.saturating_add(share);
+		}
+
+		assert_eq!(orml_rewards::PoolInfos::<T>::get(&pool_id).total_shares, total_share);
```
Comment (shaunxw) [inline modules/incentives/src/migration.rs:101]: I think we also need `assert_eq` to check `rewards` of each `PoolInfo` is empty btree map.

**Review by shaunxw — APPROVED** [review]

File: runtime/karura/src/lib.rs [hunk 3]
```diff
@@ -1799,14 +1799,8 @@ pub type SignedPayload = generic::SignedPayload<Call, SignedExtra>;
...
-pub type Executive = frame_executive::Executive<
-	Runtime,
-	Block,
-	frame_system::ChainContext<Runtime>,
-	Runtime,
-	AllPalletsWithSystem,
-	module_incentives::migration::ClearDexSavingRewardRates<Runtime>,
->;
+pub type Executive =
+	frame_executive::Executive<Runtime, Block, frame_system::ChainContext<Runtime>, Runtime, AllPalletsWithSystem, ()>;
```
Comment (xlc) [inline runtime/karura/src/lib.rs:1803]: migration is not yet executed on karura so I think we should leave this until it is executed

**Review by xlc — APPROVED** [review]

**Review by shaunxw — APPROVED** [review]

**Pull request merged**